### PR TITLE
PR for the setting up the develop branch as main branch

### PR DESCRIPTION
**Pull request by shantanuFyndGit**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,40 @@
-# pixelbin-figma-plugin
\ No newline at end of file
+Below are the steps to get your plugin running. You can also find instructions at:
```
Comment (sagarika-fynd) [inline README.md:1]: re-write this completely and make it more relevant

File: formOptions.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const ebgOptions = [
```
Comment (sagarika-fynd) [inline formOptions.js:1]: rename file to, constants.js and name variable as erasebgOptions

File: manifest.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{
+	"name": "14thDec",
```
Comment (sagarika-fynd) [inline manifest.json:2]: change the name

File: manifest.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	"id": "1317177076838052932",
+	"api": "1.0.0",
+	"main": "dist/code.js",
+	"capabilities": [],
+	"enableProposedApi": false,
+	"editorType": ["figma"],
+	"ui": "ui.html",
+	"networkAccess": {
+		"allowedDomains": [
+			"https://cdn.pixelbin.io/v2/documentation/original/pixelbin.v5.4.0.js",
```
Comment (sagarika-fynd) [inline manifest.json:12]: remove all the urls no longer required

File: package.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
+{
+	"name": "14thDec",
```
Comment (sagarika-fynd) [inline package.json:2]: change name

File: package.json [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+	"version": "1.0.0",
+	"description": "Your Figma Plugin",
```
Comment (sagarika-fynd) [inline package.json:4]: write clear description as well

File: src/apiCall.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
+// import { loadSignatureHeaders } from "./interceptors";
+type Options = {
```
Comment (sagarika-fynd) [inline src/apiCall.ts:2]: move to a folder common

File: src/code.ts [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+figma.showUI(__html__, {
+	title: "Erase Bg",
+	height: 400,
+	width: 248,
+	themeColors: true,
+});
+
+figma.ui.onmessage = async (msg) => {
+	if (msg.type === "initialCall") {
```
Comment (sagarika-fynd) [inline src/code.ts:12]: move initalCall to an ENUM

File: src/code.ts [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+		const body = {
+			type: "createForm",
```
Comment (sagarika-fynd) [inline src/code.ts:14]: same for the createForm ,  savedFormValue , 	if (msg.type === "transform") {
 etc

File: src/interceptors/index.ts [hunk 10]
```diff
@@ -0,0 +1,159 @@
+import { isAbsoluteURL, combineURLs } from "./helpers";
```
Comment (sagarika-fynd) [inline src/interceptors/index.ts:1]: move interceptor to common function

File: ui.html [hunk 11]
```diff
@@ -0,0 +1,259 @@
...
+			parent.postMessage(
+				{
+					pluginMessage: {
+						type: "initialCall",
+					},
+				},
+				"*"
+			);
+
+			function camelCase(str) {
```
Comment (sagarika-fynd) [inline ui.html:133]: move to a util class

File: ui.html [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+				return str
+					.replace(/(?:^\w|[A-Z]|\b\w)/g, function (word, index) {
+						return index == 0 ? word.toLowerCase() : word.toUpperCase();
+					})
+					.replace(/\s+/g, "");
+			}
+
+			window.onmessage = async (event) => {
+				const { data } = event;
+				if (data.pluginMessage.type === "createForm") {
```
Comment (sagarika-fynd) [inline ui.html:143]: enum

File: ui.html [hunk 13]
```diff
@@ -0,0 +1,259 @@
...
+					optionsArray = data.pluginMessage.optionsArray;
+					prevSavedFormValues = data.pluginMessage.savedFormValue;
+					createOptionsForm(false);
+				}
+				if (data.pluginMessage.type === "selectedImage") {
+					bytes = data.pluginMessage.imageBytes;
+					// console.log("bytes", data.pluginMessage.imageBytes);
+					const pixelbin = new Pixelbin({
+						cloudName: "muddy-lab-41820d",
```
Comment (sagarika-fynd) [inline ui.html:152]: move to config

File: ui.html [hunk 14]
```diff
@@ -0,0 +1,259 @@
...
+						"__playground/playground-default.jpeg"
+					);
+					let t = Pixelbin.transformations.EraseBG.bg();
+					image.setTransformation(t);
+					// console.log("url after remoing bg", image.getUrl());
+				}
+			};
+
+			function createOptionsForm(isReset) {
+				var formHolder = document.getElementById("formContainer");
```
Comment (sagarika-fynd) [inline ui.html:165]: rename formContainer to something more relevant

File: ui.html [hunk 15]
```diff
@@ -0,0 +1,259 @@
...
+							temp = optionsArray[foundIndex];
+
+						if (temp.type == "enum") document.getElementById(key).value = value;
+						if (temp.type == "boolean")
+							document.getElementById(key).checked = value;
+					});
+				}
+			}
+
+			document.getElementById("apply").onclick = () => {
```
Comment (sagarika-fynd) [inline ui.html:231]: change elementId

Comment (shantanuFyndGit) [reply ui.html:231]: resolved

File: ui.html (hunk 14 above)
Comment (shantanuFyndGit) [reply ui.html:165]: changes done

File: package.json (hunk 6 above)
Comment (shantanuFyndGit) [reply package.json:4]: changes done

File: package.json (hunk 5 above)
Comment (shantanuFyndGit) [reply package.json:2]: changes done

File: manifest.json (hunk 4 above)
Comment (shantanuFyndGit) [reply manifest.json:12]: changes done

File: manifest.json (hunk 3 above)
Comment (shantanuFyndGit) [reply manifest.json:2]: changes done

File: formOptions.js (hunk 2 above)
Comment (shantanuFyndGit) [reply formOptions.js:1]: changes done

File: ui.html (hunk 11 above)
Comment (shantanuFyndGit) [reply ui.html:133]: changes done

File: src/apiCall.ts (hunk 7 above)
Comment (shantanuFyndGit) [reply src/apiCall.ts:2]: Changes done

File: src/interceptors/index.ts (hunk 10 above)
Comment (shantanuFyndGit) [reply src/interceptors/index.ts:1]: Changes done

**Pull request closed without merging**